pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java

```diff
@@ -385,11 +391,38 @@ private static TransformStep newComputeFieldFunction(ComputeConfig config) {
     return ComputeStep.builder().fields(fieldList).build();
   }
 
+  @SneakyThrows
   private TransformStep newComputeAIEmbeddings(ComputeAIEmbeddingsConfig config) {
+    String targetSvc = config.getService();
+    if (targetSvc == null || Strings.isNullOrEmpty(targetSvc)) {
+      targetSvc = ComputeAIEmbeddingsConfig.SupportedServices.OPENAI.name();
+    }
+
+    ComputeAIEmbeddingsConfig.SupportedServices service =
+        ComputeAIEmbeddingsConfig.SupportedServices.valueOf(targetSvc.toUpperCase());
```

Comment:
let's add a level of smartness/autoconf here:
if the user configured only "openai" then the service here is "openai" by default,
if the user configured the HF API in the top level element, then the service is "hugging face api" by default

probably also for the "local" mode we can autodetect it somehow, it is not expected that a user uses both HG API and HG local mode in the same function.  (but it is expected that they use HG to compute embeddings and OpenAPI for Completion for instance)

Reply:
OpenAI is configured either way because it is used in other steps, hence explicit use of config.getService().